src/containers/Dashboard/DashboardContainer.tsx

```diff
@@ -11,30 +11,34 @@ import {DealsContainer} from "../Deals/DealsContainer";
 import {PledgedCollateralContainer} from "../PledgedCollateral/PledgedCollateralContainer";
 import {useSelector, useDispatch} from "react-redux";
 import {RootState} from "../../app/rootReducer";
-import {useHistory} from "react-router-dom";
-import {Routes} from "../../constants/routes";
-import {getNodeList} from "./NodeSlice";
+import {getNodeList, getDiskInfo, getGeneralInfo} from "./NodeSlice";
 
 export const DashboardContainer = (): ReactElement => {
-    const [areElementsHidden, setElementsHidden] = useState(true);
-    const history = useHistory();
-    const state = useSelector((state: RootState) => state);
+    const [areElementsHidden, setElementsHidden] = useState<boolean>(true);
+    const [selectedNodeIndex, setSelectedNodeIndex] = useState<number>(0);
     const dispatch = useDispatch();
+    const state = useSelector((state: RootState) => state);
+    const nodeList = state.node.nodeList;
 
     useEffect(() => {
-        if (!state.user.token) {
-            history.push(Routes.LOGIN_ROUTE);
-        } else {
-            dispatch(getNodeList(state.user.token));
+        dispatch(getNodeList());
+        if (nodeList[0] && nodeList[0].id) {
+            dispatch(getGeneralInfo(nodeList[selectedNodeIndex].id));
+            dispatch(getDiskInfo(nodeList));
```

Comment:
getDiskInfo should be fetching only 1 disk info and here you send list :) Use proper naming.

Reply:
But we need to display a list of disk data on NodeListContainer, Free/Taken space.
That's why getDiskInfo() takes array and loops to get every disk info of available nodes.

https://www.figma.com/file/EqJSrELwuL0NUS7fQv4TXP/Hactar?node-id=245%3A291

Reply:
I am aware of that, I am just referring to the naming